src/track-menu-item.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+
+document.addEventListener( 'DOMContentLoaded', function () {
+	// Track clicks on the "Upgrades > Plans" menu item in the sidebar
+	const el = document.querySelector(
+		'#toplevel_page_paid-upgrades > ul > li.wp-first-item'
+	);
+	if ( el ) {
+		el.addEventListener( 'click', function () {
+			recordTracksEvent( 'calypso_sidebar_item_click', {
+				path: '/plans',
+			} );
+		} );
+	}
```

Comment:
Good points! @daledupreez Thank you for the suggestion! I made some slight adjustments by moving the href check logic outside. This is because the event listener target can sometimes be a child element. Additionally, by doing this, we can only bind the necessary elements. 🙂